arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json

```diff
@@ -0,0 +1,307 @@
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to list log profiles",
+            "schema": {
+              "$ref": "#/definitions/LogProfileCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Incident": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Incident name."
+        },
+        "ruleName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Rule name that is associated with the incident."
+        },
+        "isActive": {
+          "readOnly": true,
+          "type": "boolean",
+          "description": "A boolean to indicate whether the incident is active or resolved."
+        },
+        "activatedTime": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The time at which the incident was activated in ISO8601 format."
+        },
+        "resolvedTime": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The time at which the incident was resolved in ISO8601 format. If null, it means the incident is still active."
+        }
+      },
+      "description": "An alert incident indicates the activation status of an alert rule."
+    },
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "RetentionPolicy": {
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "a value indicating whether the retention policy is enabled."
+        },
+        "days": {
+          "type": "integer",
+          "format": "int32",
+          "description": "the number of days for the retention."
+        }
+      },
+      "required": [ "enabled", "days" ],
+      "description": "Specifies the retention policy for the log."
+    },
+    "LogProfileCreateOrUpdateParameters": {
```

Comment:
@gucalder - If the LogProfileResource inherits (allof) from the Resource and the Resource has location required then the user will have to provide the location. Why would the location be empty? Are you saying that the service does not return the location property in the response ? If that is true then that is a bug in the service.